pages/tags/[tag].tsx

```diff
@@ -33,7 +33,10 @@ export const getStaticProps: GetStaticProps<Props> = async context => {
     return {
         props: {
             tag,
-            records: filteredRecords.map(record => omitUndefinedFields({ ...record, url: `/${record.slug}` })),
+            records: filteredRecords.map(record => omitUndefinedFields({ 
+                ...record, 
+                url: record.frontMatter.type === 'posts' ? `/${record.slug}` : `/${record.frontMatter.type}/${record.slug}`
```

Comment:
This kind of special treatment for "posts" makes me wonder: what's our reasoning behind namespacing the non-posts records (guides, etc) if we aren't namespacing "posts"? Could we drop namespacing altogether? We previously referenced [this article](https://swyx.io/namespacing-site) about it

Reply:
Thanks for bringing this up! My take on #109 was that it was more of a short-term fix, since the current landing page is also functioning as the `posts` index page. But this wasn't meant to be the long-term approach, right? I totally agree with this concern, too — I would prefer to not give `posts` special treatment. Personally, I think we should take advantage of some of the things I mention in this thread https://github.com/sourcegraph/learn/pull/127#discussion_r676596190
I'm imagining that long term we have a homepage that does not function as an index for a single content type, but could instead act as a jumping off point for different content types/indices. Maybe it would be worth prioritizing that type of redesign so that we're not introducing a bunch of bespoke logic that gives `posts` special treatment?